src/main/java/buildcraftAdditions/api/networking/ISyncObject.java

```diff
@@ -11,7 +11,7 @@
  */
 public interface ISyncObject {
 
-	public ByteBuf writeToByteBuff(ByteBuf buf);
+	void writeToByteBuff(ByteBuf buf);
 
-	public ByteBuf readFromByteBuff(ByteBuf buf);
+	void readFromByteBuff(ByteBuf buf);
```

Comment:
it should work that way but when i was testing this earlier it didn't work at all, it was reading incorrect data as if it started reading from the beginning of the bytebuf each time